pkg/mover/chart.go

```diff
@@ -253,3 +276,19 @@ func saveChart(chart *chart.Chart, toChartFilename string) error {
 
 	return os.Remove(tempDir)
 }
+
+type Option func(*ChartMover)
```

Comment:
I find this construct for options to be weird. Have you seen it elsewhere?

Does not feel idiomatic to me.

Reply:
Yes, it's idiomatic, or at least I've seen it in multiple Go projects. It's called options pattern. You can see how grpc implements a similar pattern [here](https://pkg.go.dev/google.golang.org/grpc#DialOption) 

Some articles talking about it https://www.sohamkamani.com/golang/options-pattern/
https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
You are right! Yes in this case my alternative would not work as the "constructor" is doing more work than just constructing.

Reply:
Thanks for the pointer.

Reply:
I've seen it a few ways. Either with chaining options like:
```
chartMover := NewChartMover(...).WithRetries(4).WithLogger(cmd)
```

or with an options list:
```
chartMover := NewChartMover(
...,
WithRetries(4), WithLogger(cmd)
)

func NewChartMover, options ...Option) *ChartMover{
...
```